### PR TITLE
fix: crashed when removing a invalid index (-1)

### PR DESCRIPTION
**Pull request by FeiWang1119**
handle the exceptional case that location is -1

Issue: https://github.com/linuxdeepin/developer-center/issues/7964

### CODE REVIEW

File: src/dbus/applicationservice.cpp [hunk 1]
```diff
@@ -944,6 +944,11 @@ LaunchTask ApplicationService::unescapeExec(const QString &str, const QStringLis
     auto filesCode = list.first().back().toLatin1();
     auto codeStr = QString(R"(%%1)").arg(filesCode);
     auto location = execList.indexOf(codeStr);
+    if (location == -1 && execList.back().contains(codeStr)) {
```
Comment (ComixHe) [inline src/dbus/applicationservice.cpp:947]: 不能这么改，下面进行url替换的地方会用到location，Launch传url进来还是会崩溃，如果要改的话需要把改整个流程

Comment (FeiWang1119) [reply src/dbus/applicationservice.cpp]: done

File: src/dbus/applicationservice.cpp [hunk 2]
```diff
@@ -909,8 +909,95 @@ std::optional<QStringList> ApplicationService::unescapeExecArgs(const QString &s
 LaunchTask ApplicationService::unescapeExec(const QString &str, const QStringList &fields) noexcept
 {
     LaunchTask task;
-    auto opt = unescapeExecArgs(str);
 
+    QString content = str;
+    if (str.count('%') == 1) {
```
Comment (ComixHe) [inline src/dbus/applicationservice.cpp:914]: 不要这样改，因为Exec里允许%%这种转义

**Review by ComixHe — CHANGES_REQUESTED** [review]

File: src/dbus/applicationservice.cpp (hunk 2 above)
Comment (ComixHe) [reply src/dbus/applicationservice.cpp:914]: https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html

Comment (FeiWang1119) [reply src/dbus/applicationservice.cpp]: done

**Review by ComixHe — APPROVED** [review]

**Pull request merged**